src/Api/Validator.php

```diff
@@ -189,6 +189,13 @@ private function check_boolean(Shape $shape, $value)
 
     private function check_string(Shape $shape, $value)
     {
+        if ($shape['jsonvalue']
+            && !is_resource($value)
+            && (is_object($value) || is_array($value))
```

Comment:
This isn't properly validating jsonvalue. If the shape has the jsonvalue trait, then the provided value MUST be something that can be json_encoded. Therefore, you should have a branch here that is only conditional on if the jsonvalue trait is present. Then inside of that branch, you ensure that the value can be json encoded. If it cannot, than you must add an error via `addError`. Otherwise, if the value can be json encoded, then return early.

I did a quick benchmark to see the fastest method to check if a value can be json_encoded. A function that check for the appropriate types or interfaces is almost always faster than attempting to json_encode and look for errors. Here's my gist with the example functions: https://gist.github.com/mtdowling/ed5ea633ec8358e6d2d399a8c0aa1220. I essentially whitelist specific types and have a special case for object.